### ISSUE TITLE
"Debugger can not start" when using VSCode through a Load Balancer on AWS

### ISSUE THREAD

**Issue by amirsamary** [intersystems-community/vscode-objectscript#607]
Hi!

I have an IRIS mirror cluster deployed in AWS using ICM. 

It has been deployed with the DM roles with LoadBalancer=true which means that ICM deploys a load balancer for me. When using the Load balancer, it will always take me to the PRIMARY mirror member which is very convenient.

I can use VSCode connected to this cluster using the Load Balancer without issues. Everything works. I can open the management portal. I can save and compile classes.

The only thing that does not work is debugging. Every time I try to click on "Debug this method" on a classmethod, I get the error: "Debugger can not start". It then offers me to open launch.json.

The load balancer endpoint is a FQDN that looks like this: xpto-data-poc-12655442165.us-east-1.elb.amazonaws.com

I have tried many things and I think this is a bug. Why? If I stop using the the FQDN above and change it for the IP address of the PRIMARY node in my cluster (skipping the Load Balancer altogether), debugging works.

Maybe the issue has something to do with the port number? I mean: When using the LB, I use port 80, which gets redirected to 52773 in IRIS by the Load Balancer. When accessing the PRIMARY node directly, I access it through port 52773.

**Comment by isc-bsaviano** [comment]
@amirsamary The debugger uses a websocket to send messages to and receive messages from the server. How does your load balancer handle this? The initial request to the server is a REST request that upgrades to a websocket so maybe the protocol switch fails due to the load balancer. I'm not familiar with load balancers though so I could be wrong.

**Comment by amirsamary** [comment]
The load balancer can route anything. HTTP requests that keep going through the same port will just work. AFAIK, web sockets are just long running HTTP connections (it starts as HTTP and then stays connected). So this should work without issues. The question is: How is this debugger WebSocket connection opened? Is it opened using the same port configured in the settings or is it opened with the port the server is listening to? I mean: Everything works, except debugging. So what I am guessing is happening is that all requests are being sent to port 80 (load balancer); but when it is time to debug, VSCode relies on the port the server tells it (52773) instead of the port that is in the settings (80) and tries to open the WebSocket through 52773. That won't work since the LB will not expose 52773. But I am just guessing.

**Comment by isc-jamien** [comment]
No, the debugger start command is just a regular HTTP request to the same port. This will turn out to be some sort of configuration requirement on the load balancer

**Comment by amirsamary** [comment]
Ok. I changed the Loadbalancer configuration. It allows me to specify the protocol for the port. I had it with HTTP. I changed it to TCP. I continued to be able to open the management portal with this setting. I continued to be able to connect to IRIS using VSCode and compile my classes using the LB endpoint (as before). But now I can also debug.

So, because of the WebSocket, the protocol must be configured as TCP instead of HTTP.
